src/ostree/ot-builtin-size-summary.c

```diff
@@ -0,0 +1,138 @@
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Author: Vivek Dasmohapatra <vivek@collabora.com>
+ */
+
+#include "config.h"
+
+#include "ot-main.h"
+#include "ot-builtins.h"
+#include "ostree.h"
+#include "otutil.h"
+
+gboolean opt_detailed;
+
+static GOptionEntry options[] = {
+  { NULL }
+};
+
+gboolean
+ostree_builtin_size_summary (int argc, char **argv, GCancellable *cancellable, GError **error)
+{
+  GOptionContext *context;
+  OstreeRepo *repo;
+  gsize entries = 0;
+  const char *remote = NULL;
+  const char *ref = NULL;
+  gs_unref_variant GVariant *index = NULL;
+  gs_free gchar *refspec = NULL;
+  gs_free gchar *revision = NULL;
+  gint64 archived = 0;
+  gint64 unpacked = 0;
+  gsize  fetch_needed = 0;
+  gint64 new_archived = 0;
+  gint64 new_unpacked = 0;
+
+  context = g_option_context_new("[REMOTE] BRANCH - Display the summary for branch");
+  g_option_context_add_main_entries (context, options, NULL);
+
+  if (!ostree_option_context_parse (context, options, &argc, &argv, OSTREE_BUILTIN_FLAG_NONE, &repo, cancellable, error))
+    goto out;
+
+  if (!ostree_repo_open (repo, cancellable, error))
+    goto out;
+
+  switch (argc)
+    {
+    case 3:
+      remote = argv[1];
+      ref = argv[2];
+      refspec = g_strdup_printf ("%s/%s", remote, ref);
+      break;
+
+    case 2:
+      ref = argv[1];
+      refspec = g_strdup (ref);
+      break;
+
+    case 1:
+      ot_util_usage_error (context, "BRANCH must be specified", error);
+      goto out;
+
+    default:
+      ot_util_usage_error (context, "only one branch may be summarised", error);
+      goto out;
+    }
+
+  // if there's a remote, we might not have pulled the metadata yet:
+  if (!ostree_repo_resolve_rev (repo, refspec, remote != NULL, &revision, error))
+    goto out;
+
+  // nothing in the cache, but try and fetch it if it's a remote refspec:
+  if (!revision && remote)
+    {
+      OstreeRepoPullFlags flags = OSTREE_REPO_PULL_FLAGS_METADATA;
+      gchar *pullrefs[] = { (gchar *) ref, NULL };
+
+      if (!ostree_repo_pull (repo, remote, pullrefs, flags, NULL, cancellable, error))
+        goto out;
+
+      if (!ostree_repo_resolve_rev (repo, refspec, FALSE, &revision, error))
+        goto out;
+    }
+
+  if (!revision)
+    {
+      if (!*error)
```

Comment:
I don't think this needs to be checked. `error` should only be set if the previous calls to `ostree_repo_resolve_rev` failed, in which case you'll never get here.